platforms/platform_db/platform_defaults.json

```diff
@@ -59,10 +48,106 @@
                      "function of the number lines in flight, indepent of request sizes."
                   ],
-               "max-bw-active-lines-c0": "{ 768, 512, 256, 256 }",
+               "max-bw-active-lines-c0": "{ 512, 512, 256, 256 }",
                "max-bw-active-lines-c1": "{ 512, 256, 128, 128 }",
 
-               "max-outstanding-mmio-rd-reqs": 32
+               "max-outstanding-mmio-rd-reqs": 32,
+
+               "comment":
+                  [
+                     "CCI-P requires that an AFU register both the request and response",
+                     "signals before any combinational logic.  On some platforms, it",
+                     "is easier to meet timing when there is more than one register",
+                     "stage.  num-timing-reg-stages is a platform-specific recommendation",
+                     "to AFUs, suggesting the number of registered CCI-P request and",
+                     "response stages for timing across the partial reconfiguration boundary.",
+                     "",
+                     "Note: There is no extra platform-side buffering added for handling",
+                     "extra almost full signals!  Extra buffering beyond one stage",
+                     "counts against the sending limit following almost full.  Typically,",
+                     "2 * (num-timing-reg-stages - 1) slots are lost to buffering.",
+                     "This accounts for the added latency of receiving almost full and",
+                     "also accounts for extra in-flight requests beyond the CCI-P",
+                     "allotted single stage."
+                  ],
+               "num-timing-reg-stages": 1,
+
+               "comment":
+                  [
+                     "This field requests that the platform register CCI-P requests",
+                     "and responses on the AFU side of the partial reconfiguration",
+                     "boundary.  Register stages injected here count against the",
+                     "almost full sending limit in both directions!",
+                     "See num-timing-reg-stages above for a discussion.",
+                     "",
+                     "In addition to integer counts, the string 'auto' (use double",
+                     "quotes) will request (num-timing-reg-stages - 1) auto-inserted",
+                     "stages.  One is subtracted because the AFU is still expected",
+                     "to register the CCI-P signals.  Automatic mode shifts",
+                     "responsibility for managing variations in num-timing-reg-stages",
+                     "from AFUs to the platform."
+                  ],
+               "num-auto-timing-reg-stages": null,
+
+               "comment":
+                  [
+                     "AFUs may request that the platform transform the CCI-P clock",
+                     "to a different clock domain by setting this value to the",
+                     "name of another incoming clock."
```

Comment:
an example in DCP would be DDR clock? Do they use the clock net name at the PR boundary? Perhaps clarify what name they need to use for the clock?

Reply:
There isn't currently a name for the DDR clock. I thought about that, but it would be difficult. For now, you can pick any of the standard CCI-P clocks.

The other problem with using a DDR clock is there is no guarantee that all the DDR clocks from different banks are the same!

Reply:
Ok. So, the legal values are "pClkDiv2" and "pClkDiv4" and "pClk", default being pClk. I wonder if you want to mention that in the comment for clarity?

Reply:
And the user clocks so you can have any frequency. Ok, I will.